src/gui/DatabaseWidget.cpp

```diff
@@ -216,6 +216,7 @@ DatabaseWidget::DatabaseWidget(QSharedPointer<Database> db, QWidget* parent)
     connect(m_csvImportWizard, SIGNAL(importFinished(bool)), SLOT(csvImportFinished(bool)));
     connect(this, SIGNAL(currentChanged(int)), SLOT(emitCurrentModeChanged()));
     connect(this, SIGNAL(requestGlobalAutoType(const QString&)), parent, SLOT(performGlobalAutoType(const QString&)));
+    connect(this, SIGNAL(updateLastDatabases(const QString&)), parent, SLOT(updateLastDatabases(const QString&)));
```

Comment:
This is an anti-pattern. You should always connect from parent to child, not child to parent. This should be in DatabaseTabWidget

Reply:
Ok, I moved the connect to the `DatabaseTabWidget` class.
Actually this solution does not work, for some reason the `filePath` is empty. May be I missed something?